### DISCUSSION TITLE
How does it affect performance

### DISCUSSION THREAD

**Discussion by eatenpancreas** [makspll/bevy_mod_scripting#151] [General]
Hi, i'm looking into adding this crate to my game for modding early on, and i'm wondering how much overhead it would generally be to write the majority of 'business logic' in Lua. Does anyone have any info on this? Is there any major difference between languages?

I know it's a very broad question with a "it depends on the game", so i dont expect any concrete answers. But is it viable for a game with an already heavy CPU load? One with potentially lots of events. Or should i look somewhere else if i want to add modding at higher performance

**Comment by makspll** [comment]
Hey, I don't have a very good answer to this at the moment, see #6, the library is very much not production ready yet. 

`readme.md`
>  It's important to note that this is a work in progress and not yet optimized or complete

I suspect you'll see a good bit of overhead due to how accessing values from lua works at the moment:
- Each reference is a pair of base reflection ID + reflection path
- Every time we access something say `comp.my_vec[2]` we need to:
    - perform lookups in the type registry at each step of the reflection path, for lua conversion functions
    - grab mutexes on the resource/component/allocation ID to prevent breaking rust aliasing rules
    - convert between the lua and rust types

**Reply by makspll** [reply]
Benchmarks will now be published under: https://bencher.dev/perf/bms/plots

I believe reflection is not actually too bad performance wise, querying will be worse according to the initial data
